boring/src/ssl/connector.rs

```diff
@@ -189,6 +190,21 @@ impl ConnectConfiguration {
         self.verify_hostname = verify_hostname;
     }
 
+    /// Returns an `Ssl` configured to connect to the provided domain.
+    ///
+    /// The domain is used for SNI (if it is not an IP address) and hostname verification if enabled.
+    pub fn into_ssl(mut self, domain: &str) -> Result<Ssl, ErrorStack> {
+        if self.sni && domain.parse::<IpAddr>().is_err() {
+            self.ssl.set_hostname(domain)?;
+        }
+
+        if self.verify_hostname {
+            setup_verify_hostname(&mut self.ssl, domain)?;
+        }
```

Comment:
Those checks are different than in `connect` below, they should be the same and `connect` should call `into_ssl`.

Reply:
Updated. I keep the `&& domain.parse::<IpAddr>().is_err()` which is a fix ported from openssl-rs.